### PR TITLE
Always treat encoding as a string for encoding.trim()

### PR DESCRIPTION
**Pull request by COBOL-Erik**
encoding.trim() only works if encoding is interpreted as a string, but it might just be a number.

**What It Does**

**How to Test**

**Review Checklist**
I certify that I have:
- [X] tested my changes
- [x] added/updated automated tests
- [x] updated the changelog
- [X] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

**Additional Comments**

### CODE REVIEW

File: packages/zosjobs/src/GetJobs.ts [hunk 1]
```diff
@@ -375,7 +375,10 @@ export class GetJobs {
         ImperativeExpect.toNotBeNullOrUndefined(spoolId, "Required parameter spoolId must be defined");
         let parameters: string = "/" + encodeURIComponent(jobname) + "/" + encodeURIComponent(jobid) +
             JobsConstants.RESOURCE_SPOOL_FILES + "/" + encodeURIComponent(spoolId) + JobsConstants.RESOURCE_SPOOL_CONTENT;
-        if (encoding && encoding.trim() != "") {parameters += "?fileEncoding=" + encoding;}
+        if (encoding) {
+            encoding = String(encoding);
+            if(encoding.trim() != "") {parameters += "?fileEncoding=" + encoding;}
```
Comment (zFernand0) [inline packages/zosjobs/src/GetJobs.ts:380]: What do you think about combining the two if conditions? 😋 
```suggestion
        if (encoding != null && String(encoding).trim !== "") {
            parameters += "?fileEncoding=" + encoding;
        }
```

**Review by zFernand0 — COMMENTED** [review]
Hey @COBOL-Erik,
The fix LGTM! 😋 

Curious if you could: 
- Add a Changelog entry to the `packages/zosjobs/src/CHANGELOG.md` file?
- (maybe) add a quick test for the encoding being a number? 🙏

**Comment by COBOL-Erik** [comment]
> Hey @COBOL-Erik, The fix LGTM! 😋
> 
> Curious if you could:
> 
> * Add a Changelog entry to the `packages/zosjobs/src/CHANGELOG.md` file?
> * (maybe) add a quick test for the encoding being a number? 🙏

Hi!
I added a change log entry. Not sure how I would add a test, though. Could you help me out?

File: packages/zosjobs/CHANGELOG.md [hunk 2]
```diff
@@ -2,11 +2,16 @@
 
 All notable changes to the Zowe z/OS jobs SDK package will be documented in this file.
 
+## `8.7.2`
```
Comment (zFernand0) [inline packages/zosjobs/CHANGELOG.md:5]:
```suggestion
## Recent Changes
```

File: packages/zosjobs/src/GetJobs.ts [hunk 3]
```diff
@@ -375,7 +375,10 @@ export class GetJobs {
         ImperativeExpect.toNotBeNullOrUndefined(spoolId, "Required parameter spoolId must be defined");
         let parameters: string = "/" + encodeURIComponent(jobname) + "/" + encodeURIComponent(jobid) +
             JobsConstants.RESOURCE_SPOOL_FILES + "/" + encodeURIComponent(spoolId) + JobsConstants.RESOURCE_SPOOL_CONTENT;
-        if (encoding && encoding.trim() != "") {parameters += "?fileEncoding=" + encoding;}
+        if (encoding != null && String(encoding).trim !== "") {
+            parameters += "?fileEncoding=" + encoding;
+        }
+        }
```
Comment (zFernand0) [inline packages/zosjobs/src/GetJobs.ts:381]:
```suggestion
        if (encoding != null && String(encoding).trim() !== "") {
            parameters += "?fileEncoding=" + encoding;
        }
```

**Review by zFernand0 — CHANGES_REQUESTED** [review]
Left a few comments to address.

Note: even though typescript should prevent people from calling the `getSpoolContentById` with a number e.g. `939` as the encoding value, nothing prevents people from bypassing it 😋 

And for the test, you could add a new one in:
- `packages/zosjobs/__tests__/__unit__/GetJobs.unit.test.ts`

And you could copy the following test to create a new one:
- `it("should return spool content from getSpoolContentById with encoding if z/OSMF response is mocked" ...`

And the resulting test might look something like this:
```typescript
        it("should return spool content from getSpoolContentById with encoding as a number if z/OSMF response is mocked", async () => {
            (ZosmfRestClient.getExpectString as any) = mockGetJobsStringData(GetJobsData.SAMPLE_JES_MSG_LG);
            const content = await GetJobs.getSpoolContentById(pretendSession, "MYJOB1", "JOB0123", 1, 939 as any);
            expect((ZosmfRestClient.getExpectString as any).mock.calls[0][1]).toContain("fileEncoding=939");
            expect(content).toEqual(GetJobsData.SAMPLE_JES_MSG_LG);
        });
```

**Comment by COBOL-Erik** [comment]
I have tried to do everything by the book now. I'm sure I have missed something... Where are we?

**Comment by t1m0thyj** [comment]
> I have tried to do everything by the book now. I'm sure I have missed something... Where are we?

There were 2 commits that were not properly signed off:
> Commit sha: [2d7b68d](https://github.com/zowe/zowe-cli/pull/2392/commits/2d7b68db69fa392eb0a97eb00f20c8564e3d7067), Author: S38453, Committer: S38453; The sign-off is missing.
> Commit sha: [5ae8fbc](https://github.com/zowe/zowe-cli/pull/2392/commits/5ae8fbc8d72193bf2b46e386aebf660ea73acd7e), Author: S38453, Committer: S38453; Expected "S38453 [erik.weyler@seb.se](mailto:erik.weyler@seb.se)", but got "COBOL-Erik [32245335+COBOL-Erik@users.noreply.github.com](mailto:32245335+COBOL-Erik@users.noreply.github.com)".

I've rebased the PR to amend these commits so the DCO check should be passing now. As the Audit check is failing on master branch, you don't need to worry about it. 🙂 

In the future, please ensure when making Git commits that your Git author is set to `COBOL-Erik` (not S38453), and either pass the `--signoff` flag on the `git commit` command or include the "Signed-off-by" line in your commit message.

**Review by t1m0thyj — APPROVED** [review]
LGTM, thanks @COBOL-Erik!

**Review by traeok — APPROVED** [review]
Thanks for the fix @COBOL-Erik!

**Pull request merged**